terraform/modules/azure-jenkinsinfra-kubernetes-agents/main.tf

```diff
@@ -0,0 +1,121 @@
+####################################################################################
+data "azurerm_resource_group" "kubernetes_agents_vnet" {
+  name = var.kubernetes_agents_network_rg_name
+}
+data "azurerm_virtual_network" "kubernetes_agents" {
+  name                = var.kubernetes_agents_network_name
+  resource_group_name = data.azurerm_resource_group.kubernetes_agents_vnet.name
+}
+data "azurerm_subnet" "kubernetes_agents" {
+  name                 = var.kubernetes_agents_subnet_name
+  virtual_network_name = data.azurerm_virtual_network.kubernetes_agents.name
+  resource_group_name  = data.azurerm_resource_group.kubernetes_agents_vnet.name
+}
+
+####################################################################################
+## Network Security Group and rules
+####################################################################################
+resource "azurerm_network_security_group" "kubernetes_agents" {
+  name                = local.nsg_agent_short_stripped_name
+  location            = data.azurerm_resource_group.kubernetes_agents_vnet.location
+  resource_group_name = var.controller_rg_name
+  tags                = var.default_tags
+}
+resource "azurerm_subnet_network_security_group_association" "kubernetes_agents" {
+  subnet_id                 = data.azurerm_subnet.kubernetes_agents.id
+  network_security_group_id = azurerm_network_security_group.kubernetes_agents.id
+}
+## Outbound Rules (different set of priorities than Inbound rules) ##
+#trivy:ignore:azure-network-no-public-egress
+resource "azurerm_network_security_rule" "allow_outbound_ssh_from_kubernetes_agents_to_internet" {
+  name                        = "allow-out-ssh-from-${local.nsg_agent_short_stripped_name}-to-internet"
+  priority                    = 4092
+  direction                   = "Outbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  source_address_prefixes     = data.azurerm_subnet.kubernetes_agents.address_prefixes
+  destination_port_range      = "22"
+  destination_address_prefix  = "Internet" # TODO: restrict to GitHub IPs from their meta endpoint (subsection git) - https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-githubs-ip-addresses
+  resource_group_name         = var.controller_rg_name
+  network_security_group_name = azurerm_network_security_group.kubernetes_agents.name
+}
+resource "azurerm_network_security_rule" "allow_outbound_jenkins_from_kubernetes_agents_to_controller" {
+  name                    = "allow-out-jenkins-from-${local.service_short_stripped_name}-to-ctrl"
+  priority                = 4093
+  direction               = "Outbound"
+  access                  = "Allow"
+  protocol                = "Tcp"
+  source_port_range       = "*"
+  source_address_prefixes = data.azurerm_subnet.kubernetes_agents.address_prefixes
+  destination_port_ranges = [
+    "80",    # HTTP for inbound websocket
```

Comment:
is this actually needed?

Reply:
I don't think so good catch. I've removed it 👍